onnxruntime/core/platform/posix/env.cc

```diff
@@ -116,31 +116,45 @@ class PosixEnv : public Env {
     return getpid();
   }
 
-  static common::Status ReadBinaryFile(int fd, const char* fname, const struct stat& stbuf, void*& p, size_t& len,
+  static common::Status ReadBinaryFile(int fd, off_t offset, const char* fname, void*& p, size_t len,
                                        OrtCallback& deleter) {
-    std::unique_ptr<char[]> buffer(reinterpret_cast<char*>(malloc(stbuf.st_size)));
+    std::unique_ptr<char[]> buffer(reinterpret_cast<char*>(malloc(len)));
     char* wptr = reinterpret_cast<char*>(buffer.get());
-    auto length_remain = stbuf.st_size;
+    auto length_remain = len;
     do {
       size_t bytes_to_read = length_remain;
-      ssize_t bytes_readed = read(fd, wptr, bytes_to_read);
+      ssize_t bytes_readed = offset > 0 ? pread(fd, wptr, bytes_to_read, offset) : read(fd, wptr, bytes_to_read);
       if (bytes_readed <= 0) {
         int err = errno;
-        (void)close(fd);
         return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "read file '", fname, "' fail, error code = ", err);
       }
       assert(static_cast<size_t>(bytes_readed) <= bytes_to_read);
       wptr += bytes_readed;
       length_remain -= bytes_readed;
     } while (length_remain > 0);
     p = buffer.release();
-    len = stbuf.st_size;
     deleter.f = DeleteBuffer;
     deleter.param = p;
     return Status::OK();
   }
 
-  common::Status ReadFileAsString(const char* fname, void*& p, size_t& len, OrtCallback& deleter) const override {
+  static bool CalcFileSize(int fd, size_t& len){
```

Comment:
Random nitpick, but should it just be called GetFileSize? It doesn't really calculate anything, it just gets the value.